### ISSUE TITLE
Cmdable interface has unexported functions

### ISSUE THREAD

**Issue by rogerclotet** [go-redis/redis#367]
This interface would be perfect for dependency injection or to enable mocking of a client, but it has unexported functions and that makes mocking nearly impossible.

When you create a new struct that implements the interface for mocking, the compiler complains about the unexported functions not having the expected type and the mock not implementing redis.Cmdable.

The compiler error is something like
```
cannot use client (type *"github.com/my/project/mocks".Cmdable) as type "gopkg.in/redis.v4".Cmdable in argument to NewRedis:

	*"github.com/my/project/mocks".Cmdable does not implement "gopkg.in/redis.v4".Cmdable (missing "gopkg.in/redis.v4".bitOp method)

		have "github.com/my/project/mocks".bitOp(string, string, ...string) *"gopkg.in/redis.v4".IntCmd

		want "gopkg.in/redis.v4".bitOp(string, string, ...string) *"gopkg.in/redis.v4".IntCmd
```

If I missed anything and there is a better way of mocking Client/Cmdable please let me know.

**Comment by vmihailenco** [comment]
Fixed in v4.1.9. Thanks for the report!

**Comment by rogerclotet** [comment]
There are still some unexported functions, I'm opening a pull request now. Thanks for the fast response!
